dev/Dockerfile-ubuntu-16.04

```diff
@@ -2,13 +2,21 @@
 # installs the rocm-dev package.
 
 FROM ubuntu:16.04
-LABEL maintainer=kent.knox@amd
+LABEL maintainer=peng.sun@amd
 
 # Register the ROCM package repository, and install rocm-dev package
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends curl \
   && curl -sL http://repo.radeon.com/rocm/apt/debian/rocm.gpg.key | apt-key add - \
   && printf "deb [arch=amd64] http://repo.radeon.com/rocm/apt/debian/ xenial main" | tee /etc/apt/sources.list.d/rocm.list \
   && apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-    rocm-dev \
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/*
+  sudo \
+  libelf1 \
+  build-essential \
+  git \
+  python \
+  libnuma-dev \
+  vim-nox \
+  cmake-curses-gui \
+  rocm-dev && \
```

Comment:
I don't think these are meant to be added here as this Dockerfile is meant to serve as the base according to the initial commit message.  (So this file is really just a reference.)

Modifying `rocm-terminal/Dockerfile` is correct though (although I am not familiar with rocm-terminal.)

Reply:
Thanks, I would agree to remove vim and cmake packages in the base dockerfile. The other packages were required to properly use ROCm post ROCm1.7.0 releases.

Reply:
I understand that it is needed but rocm is not meant to be installed on the base image.  I think Kent wanted to keep things modular.  The base image really suppose to be the bare minimum to have the ability to install packages.

Reply:
Okay, let me remove those additional packages and keep it as its initial purpose of design. 
Thanks.